---
title: "ITS2_analysis"
author: "EL Strand"
date: "4/3/2020"
output: html_document
---

Analysis for the ITS2 dataset associated with Strand et al. Acclimatization Dynamics: https://github.com/hputnam/Acclim_Dynamics.  

Contents:
1. Phyloseq Analysis 
2. Relative Abundance: ggplot2
3. otuSummary: Alpha Diversity 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
## select Knit > Knit Directory > Project Directory
```

#### PHYLOSEQ ANALYSIS 
Based on the tutorial from https://vaulot.github.io/tutorials/Phyloseq_tutorial.html. 
An example dataset can be found here: https://github.com/vaulot/R_tutorials/archive/master.zip 

### Phyloseq setup and reading in data files. 
Prerequisites to be installed. 
```{r}
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') # To manipulate dataframes
if ("readxl" %in% rownames(installed.packages()) == 'FALSE') install.packages('readxl') # To read Excel files into R
if ("ggplot2" %in% rownames(installed.packages()) == 'FALSE') install.packages('ggplot2') # for high quality graphics

source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")        
```

Load required libraries.
```{r}
rm(list=ls()) # clear work space
setwd("~/MyProjects/Acclim_Dynamics") # set working directory
getwd() # to check working directory

library("phyloseq")
library("ggplot2")      # graphics
library("readxl")       # necessary to import the data from Excel file
library("dplyr")        # filter and reformat data frames
```

Read in data tables. 
OTU_seq_abs is absolute abundance for sequences: "3_first_analysis_2020-04-01_12-26-53.145538.seqs.absolute.abund_only.txt";
OTU_type_abs is absolute abundance for type profiles: "3_first_analysis_2020-04-01_12-26-53.145538.profiles.absolute.abund_only.txt"
TAX was created from the list of SymPortal type outputs and clade information online: 
Samples_df was created with the different variables from the project, including treatment, tank, species, timepoint

```{r}
## relative and absolute abundance datasheets
otu_seq_abs <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Seq_Absolute_raw_input.csv", sep=",", header=TRUE) # abundance based on seqs 
otu_type_abs <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Absolute_raw_input.csv", sep=",", header=TRUE) # absolute abundance based on type 
otu_type_rel <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Relative_raw_input.csv", sep=",", header=TRUE) # relative abundance based on type 

## taxonomy and sample info datasheets
tax_seq <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Taxonomy_Seq.csv", sep=",", header=TRUE)
tax_type <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Taxonomy_Type.csv", sep=",", header=TRUE)
samples_df <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Sample_variables.csv", sep=",", header=TRUE)
samples_df <- na.omit(samples_df)

#reading timepoint as factor
samples_df$Timepoint <- as.factor(samples_df$Timepoint)

# Flipping columns and rows in the OTU matrix so that columns are sample IDs and rows are the OTU outputs
# Using transpose function: https://www.r-statistics.com/tag/transpose/. 
otu_mat_seq <- t(otu_seq_abs)
otu_mat_abs_type <- t(otu_type_abs)
otu_mat_rel_type <- t(otu_type_rel)

# deleting top row generated by transpose function
# otu_mat = otu_mat[-1,] # but that gets rid of the transpose function to flip the rows and columns... hm..

# Exporting new dataframe to get a list of the type profiles
write.csv(otu_mat_seq, "~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Seq_Absolute_transposed.csv")
write.csv(otu_mat_abs_type, "~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Absolute_transposed.csv")
write.csv(otu_mat_rel_type, "~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Relative_transposed.csv")
```
Manually delete first row (V1, V2, ..) of transposed.csv files - this is annoying, come back to this later. 

```{r}
otu_seq_transposed <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Seq_Absolute_transposed.csv", sep = ",", header = TRUE)
otu_type_abs_transposed <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Absolute_transposed.csv", sep = ",", header = TRUE)
otu_type_rel_transposed <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Relative_transposed.csv", sep = ",", header = TRUE)

# take out the "X" in front of column names 
for (col in 1:ncol(otu_seq_transposed)) {
  colnames(otu_seq_transposed)[col] <- sub("X", "", colnames(otu_seq_transposed)[col])
}
for (col in 1:ncol(otu_type_abs_transposed)) {
  colnames(otu_type_abs_transposed)[col] <- sub("X", "", colnames(otu_type_abs_transposed)[col])
}
for (col in 1:ncol(otu_type_rel_transposed)) {
  colnames(otu_type_rel_transposed)[col] <- sub("X", "", colnames(otu_type_rel_transposed)[col])
}

## come back to tying to take out the "X" in the row names of otu_type_transposed. Until then, the Type profile names in the taxonomy file need to match these transposed files.
```

Transforming the dataframes. 
```{r}
row.names(otu_seq_transposed) <- otu_seq_transposed$sample_uid # defining row names from the uid column
otu_seq_transposed <- otu_seq_transposed %>% select (-sample_uid) # removing the sample uid column since it is now used as a row name 

row.names(otu_type_abs_transposed) <- otu_type_abs_transposed$sample_uid # defining row names from the uid column
otu_type_abs_transposed <- otu_type_abs_transposed %>% select (-sample_uid) # removing the sample uid column since it is now used as a row name

row.names(otu_type_rel_transposed) <- otu_type_rel_transposed$sample_uid # defining row names from the uid column
otu_type_rel_transposed <- otu_type_rel_transposed %>% select (-sample_uid) # removing the sample uid column since it is now used as a row name

row.names(tax_seq) <- tax_seq$Type_Profile 
tax_seq <- tax_seq %>% select (-Type_Profile)
row.names(tax_type) <- tax_type$Type_Profile 
tax_type <- tax_type %>% select (-Type_Profile)

row.names(samples_df) <- samples_df$sample_uid
samples_df <- samples_df %>% select (-sample_uid)

# Transforming the OTU and tax datatables into matrixes 
otu_seq_transposed <- as.matrix(otu_seq_transposed)
otu_type_abs_transposed <- as.matrix(otu_type_abs_transposed)
otu_type_rel_transposed <- as.matrix(otu_type_rel_transposed)
tax_seq <- as.matrix(tax_seq)
tax_type <- as.matrix(tax_type)

# Transforming into phyloseq objects
OTU_Seq = otu_table(otu_seq_transposed, taxa_are_rows = TRUE)
OTU_Abs_Type = otu_table(otu_type_abs_transposed, taxa_are_rows = TRUE)
OTU_Rel_Type = otu_table(otu_type_rel_transposed, taxa_are_rows = TRUE)
TAX_Seq = tax_table(tax_seq)
TAX_Type = tax_table(tax_type)
samples = sample_data(samples_df)

# Double checking that the sample components match
sample_names(TAX_Seq)
sample_names(TAX_Type)
sample_names(OTU_Rel_Type)
sample_names(OTU_Abs_Type)
sample_names(OTU_Seq)
sample_names(samples)
```

Transforming into phyloseq objects. 
```{r}
carbom_Seq <- phyloseq(OTU_Seq, TAX_Seq, samples)
carbom_Abs_Type <- phyloseq(OTU_Abs_Type, TAX_Type, samples)
carbom_Rel_Type <- phyloseq(OTU_Rel_Type, TAX_Type, samples)
```

Output:
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 228 taxa and 253 samples ]
sample_data() Sample Data:       [ 253 samples by 7 sample variables ]
tax_table()   Taxonomy Table:    [ 228 taxa by 6 taxonomic ranks ]

Double checking output from carbom objects. 
```{r}
sample_names(carbom_Seq)
sample_names(carbom_Abs_Type)
sample_names(carbom_Rel_Type)
# output should be sample names (61-315)

rank_names(carbom_Seq)
rank_names(carbom_Abs_Type)
rank_names(carbom_Rel_Type)
# output should be Domain - Type categories 

sample_variables(carbom_Seq)
sample_variables(carbom_Abs_Type)
sample_variables(carbom_Rel_Type)
# output should be the experimental categories 
```

Normalize number of reads in each sample using median sequencing depth 
**Fill in explanation of what this is later**

```{r}
## Seq
total_seq = median(sample_sums(carbom_Seq))
standf_seq = function(x, t=total_seq) round(t * (x/sum(x)))
carbom_Seq = transform_sample_counts(carbom_Seq, standf_seq)

## Abs Type
total_type = median(sample_sums(carbom_Abs_Type))
standf_type = function(x, t=total_type) round(t * (x/sum(x)))
carbom_Abs_Type = transform_sample_counts(carbom_Abs_Type, standf_type)
```

## Phyloseq setup and reading in data files troubleshooting:
Error: Error in validObject(.Object) : invalid class “phyloseq” object: Component sample names do not match. Try sample_names()
Solution: The sample values likely do not between datasheets "samples" and "OTU". The columns should be the sample IDs and the rows should be the OTU outputs (Symbiont type). 

### Bar graphs 
```{r}
# filtering by the most abundance Profile Types
carbom_abund_seq <- filter_taxa(carbom_Seq, function(x) sum(x > total_seq*0.10) > 0, TRUE)

## Seq
pdf("~/MyProjects/Acclim_Dynamics/ITS2/Output/Seq_Abs_treatments.pdf")
plot_bar(carbom_abund_seq, x="Timepoint", fill="Type", facet_grid = Host_species~Treatment) + 
  geom_bar(aes(color=Type, fill=Type), stat="identity", position="stack")
dev.off()

## Type Absolute Abundance
pdf("~/MyProjects/Acclim_Dynamics/ITS2/Output/Type_Abs_treatment.pdf")
plot_bar(carbom_Abs_Type, x="Timepoint", fill="Type", facet_grid = Host_species~Treatment) + 
  geom_bar(aes(color=Type, fill=Type), stat="identity", position="stack")
dev.off()

## Type Relative Abundance
pdf("~/MyProjects/Acclim_Dynamics/ITS2/Output/Type_Rel_treatment.pdf")
plot_bar(carbom_Rel_Type, x="Timepoint", fill="Type", facet_grid = Host_species~Treatment) + 
  geom_bar(aes(color=Type, fill=Type), stat="identity", position="stack")
dev.off()
```

Using ggplot instead of phyloseq objects
```{r}
specie <- c(rep("sorgho" , 3) , rep("poacee" , 3) , rep("banana" , 3) , rep("triticum" , 3) )
condition <- rep(c("normal" , "stress" , "Nitrogen") , 4)
value <- abs(rnorm(12 , 0 , 15))
data <- data.frame(specie,condition,value)

# using otu_type_rel as beginning dataframe
library(dplyr)
library(tidyr)
library(gridExtra)
type_relative <- gather(otu_type_rel, "Type", "value", 2:13)
names(type_relative)[2] <- "Type_Profile"

taxonomy <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Taxonomy_Type.csv", sep=",", header=TRUE)
tax_sampl <- full_join(type_relative, taxonomy, by="Type_Profile") # joining taxonomy and sample info datasheets by Type Profile

samp_info <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Sample_variables.csv", sep=",", header=TRUE)
samp_info <- na.omit(samp_info)

relative_abundance_info <- full_join(tax_sampl, samp_info, by="sample_uid" ) # joining the above datasheet with the relative abundance data
relative_abundance_info <- na.omit(relative_abundance_info) # taking out the NAs at the bottom of the dataset from control mixes
relative_abundance_info$Timepoint <- as.factor(relative_abundance_info$Timepoint) # reading timepoint as a factor

M.cap.rel <- subset(relative_abundance_info, Host_species=="Montipora capitata") # subsetting the dataset to just Montipora values
P.acuta.rel <- subset(relative_abundance_info, Host_species=="Pocillopora acuta") # subsetting the dataset to just Pocillopora values

Poc.rel <- ggplot(P.acuta.rel, aes(fill=Type, y=value, x=Timepoint)) + geom_bar(position="fill", stat="identity") + 
  facet_wrap(~Treatment) + ylab("Relative Abundance") + ggtitle("Pocillopora acuta")

Mcap.rel <- ggplot(M.cap.rel, aes(fill=Type, y=value, x=Timepoint)) + geom_bar(position="fill", stat="identity") + 
  facet_wrap(~Treatment) + ylab("Relative Abundance") + ggtitle("Montipora Capitata")

Relative.Figs <- arrangeGrob(Mcap.rel, Poc.rel, ncol=1)
ggsave(file="~/MyProjects/Acclim_Dynamics/ITS2/Output/Relative.Figs.pdf", Relative.Figs, width = 11, height = 9, units = c("in"))

```


```{r}
sample_date <- merge_samples(carbom, "Timepoint")
plot_bar(sample_date, fill="Species") +
  geom_bar(aes(color=Genus, fill=Genus), stat="identity", position="stack")
```


### Heatmaps 
```{r}
# filtering by the most abundance Profile Types
carbom_abund <- filter_taxa(sample_date, function(x) sum(x > total*0.20) > 0, TRUE)
carbom_abund

plot_heatmap(carbom_abund, method = "NMDS", distance = "bray")
```

### Alpha diversity 
```{r}
plot_richness(carbom, measures = c("Chao1", "Shannon"), x="Timepoint", color = "Treatment")
```

#### RELATIVE ABUNDANCE BAR CHARTS: GGPLOT2
```{r}
Type_Profiles_Rel <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2/R_Input/Type_Relative_raw_input.csv", sep=",", header=TRUE)

```







#### ALPHA DIVERSITY USING OTUSUMMARY

### Calculating alpha diversity: https://rdrr.io/cran/otuSummary/man/alphaDiversity.html
The dataframe input needs to be a specific matrix format. See sections prior to "Transforming to phyloseq objects" and what goes into the output csv "otu_transposed". The output of this function is allBio (alpha diversity for the whole community), abundBio (alpha diversity indices for the adundant population), and rareBio (alpha diversity indices for the rare biosphere). 

Loading in dataframes and running alpha diversity metrics usings otuSummary package. 
```{r}
rm(list=ls()) # clear work space
if ("otuSummary" %in% rownames(installed.packages()) == 'FALSE') install.packages('otuSummary') 
library(otuSummary)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(plyr)
library(tidyr)
library(gridExtra)

### example data loaded to compare dataframe formatting
# data(otumothur)
# test1 <- alphaDiversity(otutab = otumothur, siteInCol = TRUE,
   # taxhead = "taxonomy", threshold = 1, percent = FALSE, write = FALSE)
### 
type_abundance_raw <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2_Data/Type_abundance_raw.csv", sep=",", header=TRUE)
type_abundance <- t(type_abundance_raw)
write.csv(type_abundance, "~/MyProjects/Acclim_Dynamics/ITS2_Data/Type_Abundance.csv")
```
Manually delete the first row V1, V2, V3 - this is annoying come back to this later.

```{r}
# Taking out the "X" in front of each column and row 
type_abundance_transposed <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2_Data/Type_Abundance.csv", sep = ",", header = TRUE)
# come back to figuring out to take out the X in row names 
for (col in 1:ncol(type_abundance_transposed)) {
  colnames(type_abundance_transposed)[col] <- sub("X", "", colnames(type_abundance_transposed)[col])
}

row.names(type_abundance_transposed) <- type_abundance_transposed$sample_uid # defining row names from the uid column
type_abundance_transposed <- type_abundance_transposed %>% select (-sample_uid) # removing the sample uid column since it is now used as a row name

# Transforming the OTU and tax datatables into matrixes 
type_abundance_transposed <- as.matrix(type_abundance_transposed)
```

```{r}
alpha <- alphaDiversity(type_abundance_transposed, siteInCol = TRUE, taxhead = NULL, threshold = 1, percent = FALSE, write = TRUE)
write.csv(alpha$allBio, "~/MyProjects/Acclim_Dynamics/ITS2_Data/alpha_allBio.csv")
write.csv(alpha$abundBio, "~/MyProjects/Acclim_Dynamics/ITS2_Data/alpha_abundBio.csv")
write.csv(alpha$rareBio, "~/MyProjects/Acclim_Dynamics/ITS2_Data/alpha_rareBio.csv")
```

Merging the diversity measures from allBio and sample information. 
```{r}
allBio <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2_Data/alpha_rareBio.csv", sep = ",", header = TRUE)
sample_info <- read.csv("~/MyProjects/Acclim_Dynamics/ITS2_Data/Sample_variables.csv", sep=",", header=TRUE)
names(allBio)[1] <- "sample_uid" # changing the column header to match that in sample_info

alpha_diversity_total <- full_join(allBio, sample_info, by="sample_uid")
```

Calculating the alpha diversity means.
```{r}
# Alpha Diversity means 
## THIS NEEDS TO BE TIMEPOINT B/C 3 TIMEPOINTS ON 20180922
alpha_diversity_total <- na.omit(alpha_diversity_total)
alpha_diversity_total$Sample_Date <- as.factor(alpha_diversity_total$Sample_Date)
alpha_diversity_total$Timepoint <- as.factor(alpha_diversity_total$Timepoint) 

# Nevermind on the below for now 
# Sample #138 has a value of 0 for shannon index - look into later; take out sample uid 77 which is sample 138
# alpha_diversity_total <- filter(alpha_diversity_total, !sample_uid==77)

Shannon.Means <- ddply(alpha_diversity_total, c('Timepoint','Host_species', 'Treatment'), summarize,
                  Shannon.mean= mean(shannon, na.rm=T), #mean shannon
                  N = sum(!is.na(shannon)), # sample size
                  Shannon.se = sd(shannon, na.rm=T)/sqrt(N)) #SE
```

Plotting alpha diversity values over timepoint, treatment, and species. 
```{r}
Mcap_alpha <- subset(Shannon.Means, Host_species=="Montipora capitata")
Mcap.A <- ggline(Mcap_alpha, x = "Timepoint", y = "Shannon.mean", color = "Treatment",
       ylim=c(0,4), 
       add = c("Shannon.se"),
       palette = c("blue", "light blue", "salmon", "red3")) + ggtitle("Montipora capitata") + ylab("Alpha diversity")
Mcap.A

Pacuta_alpha <- subset(Shannon.Means, Host_species=="Pocillopora acuta")
Pacuta.A <- ggline(Pacuta_alpha, x = "Timepoint", y = "Shannon.mean", color = "Treatment",
       ylim=c(0,3), 
       add = c("Shannon.se"),
       palette = c("blue", "light blue", "salmon", "red3")) + ggtitle("Pocillopora acuta") + ylab("Alpha diversity")
Pacuta.A

Alpha.Figs <- arrangeGrob(Mcap.A, Pacuta.A, ncol=1)
ggsave(file="~/MyProjects/Acclim_Dynamics/ITS2_Data/Output/Alpha.Figs.pdf", Alpha.Figs, width = 11, height = 6, units = c("in"))
```

Statistical analyses 
```{r}
hist(Mcap_alpha$shannon)
library(car)
shapiro.test(Mcap_alpha$shannon) 
bartlett.test(shannon~Treatment, data=Mcap_alpha)
bartlett.test(shannon~Origin, data=Mcap_alpha)

Mcap_alpha.aov <- aov(Shannon.mean ~ Treatment, data = Mcap_alpha)
summary(Mcap_alpha.aov)
TukeyHSD(Mcap_alpha.aov)

Pacuta_alpha.aov <- aov(Shannon.mean ~ Treatment, data = Pacuta_alpha)
summary(Pacuta_alpha.aov)
TukeyHSD(Pacuta_alpha.aov)
```







#### SCRIPT NOT USED B/C OF ERRORS OR DECIDED ON OTHER PACKAGES
### SCRIPT BELOW COULD BE USEFUL IN THE FUTURE 

### Calculating alpha diversity: https://www.rdocumentation.org/packages/tabula/versions/1.0.0/topics/alpha-diversity
```{r}
#dataframes - otu_mat, samples_df, tax_mat
if ("tabula" %in% rownames(installed.packages()) == 'FALSE') install.packages('tabula') 
library(tabula)

diversity(otu_mat, "shannon", simplify = FALSE)
```

### DIRICHLET REGRESSION 

```{r}
if ("DirichletReg" %in% rownames(installed.packages()) == 'FALSE') install.packages('DirichletReg') 
library(DirichletReg)
```



