# Acclimatization Dynamics Results

![expt-sch](https://github.com/hputnam/Acclim_Dynamics/blob/master/Environmental_data/HoloInt-schematic.png?raw=true)

900 corals, 450 *M. capitata* and 450 *P. acuta*, were collected from 6 sites in Kaneohe Bay, Hawai'i. These corals were randomly assigned to one of 4 treatments. 3 tanks per treatment, 12 tanks total. 2 months of treatment and 2 months of recovery in ambient conditions.   

ATAC = Ambient Temperature (27C) Ambient pCO2 (~480 uatm)  
ATHC = Ambient Temperature (27C) High pCO2 (~1,200 uatm)    
HTAC = High Temperature (29.5C) Ambient pCO2 (~480 uatm)  
HTHC = High Temperature (29.5C) High pCO2 (~1,200 uatm)

## Sampling Timeline
|                      	| Acute Stress 	|       	| Chronic Stress 	|      	|      	|      	|      	|      	|      	|      	| Recovery 	|       	|       	|       	|       	|       	|       	|       	|
|----------------------	|--------------	|-------	|----------------	|------	|------	|------	|------	|------	|------	|------	|----------	|-------	|-------	|-------	|-------	|-------	|-------	|-------	|
|                      	| Day 1        	| Day 2 	| Wk 1           	| Wk 2 	| Wk 3 	| Wk 4 	| Wk 5 	| Wk 6 	| Wk 7 	| Wk 8 	| Wk 9     	| Wk 10 	| Wk 11 	| Wk 12 	| Wk 13 	| Wk 14 	| Wk 15 	| Wk 16 	|
| Holobiont Physiology 	| x            	| x     	| x              	| x    	|      	| x    	|      	| x    	|      	| x    	|          	|       	|       	| x     	|       	|       	|       	| x     	|
| Metabolism          	| x            	| x     	| x              	| x    	|      	| x    	|      	| x    	|      	| x    	|          	|       	|       	|       	|       	|       	|       	|       	|
| Growth              	| x            	| x     	| x              	| x    	|      	| x    	|      	| x    	|      	| x    	|          	| x     	|       	| x     	|       	| x     	|       	| x     	|
| Bleaching Score     	|              	|       	| x              	| x    	| x    	| x    	| x    	| x    	| x    	| x    	| x        	| x     	| x     	| x     	| x     	| x     	| x     	| x     	|
| Survivorship        	| x            	| x     	| x              	| x    	| x    	| x    	| x    	| x    	| x    	| x    	| x        	| x     	| x     	| x     	| x     	| x     	| x     	| x     	|
| Molecular            	| 0, 6, 12 hr  	| 24 hr 	| x              	| x    	|      	| x    	|      	| x    	|      	| x    	|          	|       	|       	| x     	|       	|       	|       	| x     	|

**Response variables:**  
Molecular: Symbiont community assemblage (ITS2 Seq), bacterial community assemblage (16S Seq),  Pocillopora acuta identity (mtORF Seq), gene expression patterns (RNASeq), DNA methylation patterns (WGBS; MBD-BS).  
Holobiont physiology: protein content, total antioxidant capacity, chlorophyll concentration, tissue biomass, symbiont density.  
Metabolism: respiration and photosynthetic rates.   

## Experimental Treatments

Hobo Logger Data

![hobo](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Hobo-loggers.png?raw=true)

The sharp drops ~1-2 weeks are from removing the hobo loggers from the tanks to download the data. I need to go through and remove these time periods.

## Physiology Sample Sizes

**List of missing corals**:  
The following have been lost in transit from HIMB to URI or misplaced in a freezer in either location. I've checked URI's freezers and neighboring lab's freezers multiple times. This will decrease the sample sizes of several groups down to 2 at the smallest in Day 2 and Week 1.

| Species   	| Plug ID 	| Treatment 	| Sample Date 	|
|-----------	|---------	|-----------	|-------------	|
| Mcapitata 	| 2008    	| HTHC      	| 20180923    	|
| Mcapitata 	| 2512    	| HTHC      	| 20180923    	|
| Pacuta    	| 2522    	| HTHC      	| 20180923    	|
| Pacuta    	| 2523    	| HTHC      	| 20180923    	|
| Pacuta    	| 2752    	| ATHC      	| 20180923    	|
| Pacuta    	| 2880    	| ATHC      	| 20180923    	|
| Mcapitata 	| 2976    	| ATHC      	| 20180923    	|
| Mcapitata 	| 3003    	| ATHC      	| 20180923    	|
| Mcapitata 	| 1603    	| HTHC      	| 20180929    	|
| Pacuta    	| 1739    	| HTHC      	| 20180929    	|
| Mcapitata 	| 2200    	| HTHC      	| 20180929    	|
| Mcapitata 	| 2528    	| ATHC      	| 20180929    	|
| Mcapitata 	| 2740    	| ATHC      	| 20180929    	|
| Pacuta    	| 2859    	| HTHC      	| 20180929    	|
| Pacuta    	| 2981    	| ATHC      	| 20180929    	|
| Pacuta    	| 2984    	| ATHC      	| 20180929    	|

This table applies to respiration rates, photosynthetic rates, chlorophyll, tissue biomass, protein, cell density, and TAC. No data in recovery period for respiration and photosynthetic rates. These are all individual corals (i.e. a week 1 coral will not be in any other timepoint).   
Bleaching score, survivorship, and growth have different sample size values because corals were repetitively measured (i.e. a coral photographed in weeks 1-5 then physiologically sampled in week 6).   

| Species     	| Timepoint 	| Treatment 	| Sample Size 	| Notes                         	|
|-------------	|-----------	|-----------	|-------------	|-------------------------------	|
| M. capitata 	| Day 1     	| ATAC      	| 4           	| Not enough time in the field  	|
| M. capitata 	| Day 1     	| ATHC      	| 2           	| Not enough time in the field  	|
| M. capitata 	| Day 1     	| HTAC      	| 4           	| Not enough time in the field  	|
| M. capitata 	| Day 1     	| HTHC      	| 2           	| Not enough time in the field  	|
| P. acuta    	| Day 1     	| ATAC      	| 4           	| Not enough time in the field  	|
| P. acuta    	| Day 1     	| ATHC      	| 2           	| Not enough time in the field  	|
| P. acuta    	| Day 1     	| HTAC      	| 4           	| Not enough time in the field  	|
| P. acuta    	| Day 1     	| HTHC      	| 2           	| Not enough time in the field  	|
| M. capitata 	| Day 2     	| ATAC      	| 6           	|                               	|
| M. capitata 	| Day 2     	| ATHC      	| 4           	| Corals missing in transit     	|
| M. capitata 	| Day 2     	| HTAC      	| 6           	|                               	|
| M. capitata 	| Day 2     	| HTHC      	| 4           	| Corals missing in transit     	|
| P. acuta    	| Day 2     	| ATAC      	| 6           	|                               	|
| P. acuta    	| Day 2     	| ATHC      	| 4           	| Corals missing in transit     	|
| P. acuta    	| Day 2     	| HTAC      	| 6           	|                               	|
| P. acuta    	| Day 2     	| HTHC      	| 4           	| Corals missing in transit     	|
| M. capitata 	| 1 week    	| ATAC      	| 6           	|                               	|
| M. capitata 	| 1 week    	| ATHC      	| 4           	| Corals missing in transit     	|
| M. capitata 	| 1 week    	| HTAC      	| 6           	|                               	|
| M. capitata 	| 1 week    	| HTHC      	| 4           	| Corals missing in transit     	|
| P. acuta    	| 1 week    	| ATAC      	| 6           	|                               	|
| P. acuta    	| 1 week    	| ATHC      	| 4           	| Corals missing in transit     	|
| P. acuta    	| 1 week    	| HTAC      	| 6           	|                               	|
| P. acuta    	| 1 week    	| HTHC      	| 4           	| Corals missing in transit     	|
| M. capitata 	| 2 week    	| ATAC      	| 6           	|                               	|
| M. capitata 	| 2 week    	| ATHC      	| 6           	|                               	|
| M. capitata 	| 2 week    	| HTAC      	| 6           	|                               	|
| M. capitata 	| 2 week    	| HTHC      	| 6           	|                               	|
| P. acuta    	| 2 week    	| ATAC      	| 6           	|                               	|
| P. acuta    	| 2 week    	| ATHC      	| 6           	|                               	|
| P. acuta    	| 2 week    	| HTAC      	| 6           	|                               	|
| P. acuta    	| 2 week    	| HTHC      	| 6           	|                               	|
| M. capitata 	| 4 week    	| ATAC      	| 6           	|                               	|
| M. capitata 	| 4 week    	| ATHC      	| 6           	|                               	|
| M. capitata 	| 4 week    	| HTAC      	| 6           	|                               	|
| M. capitata 	| 4 week    	| HTHC      	| 6           	|                               	|
| P. acuta    	| 4 week    	| ATAC      	| 6           	|                               	|
| P. acuta    	| 4 week    	| ATHC      	| 6           	|                               	|
| P. acuta    	| 4 week    	| HTAC      	| 6           	|                               	|
| P. acuta    	| 4 week    	| HTHC      	| 6           	|                               	|
| M. capitata 	| 6 week    	| ATAC      	| 6           	|                               	|
| M. capitata 	| 6 week    	| ATHC      	| 6           	|                               	|
| M. capitata 	| 6 week    	| HTAC      	| 6           	|                               	|
| M. capitata 	| 6 week    	| HTHC      	| 6           	|                               	|
| P. acuta    	| 6 week    	| ATAC      	| 6           	|                               	|
| P. acuta    	| 6 week    	| ATHC      	| 6           	|                               	|
| P. acuta    	| 6 week    	| HTAC      	| 6           	|                               	|
| P. acuta    	| 6 week    	| HTHC      	| 6           	|                               	|
| M. capitata 	| 8 week    	| ATAC      	| 6           	|                               	|
| M. capitata 	| 8 week    	| ATHC      	| 6           	|                               	|
| M. capitata 	| 8 week    	| HTAC      	| 6           	|                               	|
| M. capitata 	| 8 week    	| HTHC      	| 6           	|                               	|
| P. acuta    	| 8 week    	| ATAC      	| 6           	|                               	|
| P. acuta    	| 8 week    	| ATHC      	| 6           	|                               	|
| P. acuta    	| 8 week    	| HTAC      	| 6           	|                               	|
| P. acuta    	| 8 week    	| HTHC      	| 6           	|                               	|
| M. capitata 	| 12 week   	| ATAC      	| 5           	| only 5 available to sample    	|
| M. capitata 	| 12 week   	| ATHC      	| 6           	|                               	|
| M. capitata 	| 12 week   	| HTAC      	| 6           	|                               	|
| M. capitata 	| 12 week   	| HTHC      	| 5           	| only 5 available to sample    	|
| P. acuta    	| 12 week   	| ATAC      	| 6           	|                               	|
| P. acuta    	| 12 week   	| ATHC      	| 6           	|                               	|
| P. acuta    	| 12 week   	| HTAC      	| 2           	| only 2 available to sample    	|
| P. acuta    	| 12 week   	| HTHC      	| 0           	| no corals available to sample 	|
| M. capitata 	| 16 week   	| ATAC      	| 6           	|                               	|
| M. capitata 	| 16 week   	| ATHC      	| 5           	| only 5 available to sample    	|
| M. capitata 	| 16 week   	| HTAC      	| 4           	| only 4 available to sample    	|
| M. capitata 	| 16 week   	| HTHC      	| 5           	| only 5 available to sample    	|
| P. acuta    	| 16 week   	| ATAC      	| 6           	|                               	|
| P. acuta    	| 16 week   	| ATHC      	| 6           	|                               	|
| P. acuta    	| 16 week   	| HTAC      	| 0           	| no corals available to sample 	|
| P. acuta    	| 16 week   	| HTHC      	| 0           	| no corals available to sample 	|

Not enough time in the field: We were not able to complete all of Day 1 sampling in the field because we originally included instantaneous calcification in our response variables. After Day 1 we cut out this variable, but this means we have decreased sample sizes for Day 1.

Corals missing in transit: Due to the list of missing corals, several groups on Day 2 and Week 1 will have decreased sample sizes.

Only X number available to sample: Due to lower survivorship in recovery periods, there was limited availability for sampling, resulting a smaller sample size for some groups.  

Some univariate physiology variables will also have decreased sample size groups, see each section for additional notes.

## Univariate Physiology Results   

### Survivorship

![suv](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/All_survivorship.png?raw=true)

This dataset has been QC'd by Emma; final figure as of 20210108.  
Survivorship was measured every day and each coral was noted as dead or alive for that day. This plot was made with [survfit() function](https://www.rdocumentation.org/packages/survival/versions/2.9-6/topics/survfit).

### Bleaching Score

![blch](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Photographic_Bleaching.png?raw=true)

This dataset has been QC'd by Emma; final figure as of 20210108. Corals photographed and red blue green color square quantified in ImageJ.

Week 5 data excluded because we didn't trust that set of values for any treatment. Come back to this?

### Growth

![grw](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Growth.png?raw=true)  

Buoywant weight technique

Do we like the figure format for Growth, Resp/photo rates, or AFDW/Chl?  

*P.acuta* Week 10 and 14 have the same value? This doesn't seem likely?

Figure as of 20210108.

### Respiration and photosynthetic rates

![resp](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Respirometry.png?raw=true)

No additional groups with a decreased sample size.  
This dataset has been QC'd by Emma; final figure as of 20210118.

### Tissue Biomass (AFDW)

![AFDW](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Tissue_Biomass.png?raw=true)

This dataset has been QC'd by Emma; final figure as of 20210108.

In addition to the missing list above, we won't have data for:  
- 1174: there was no homogenate tissue in this falcon tube. Aliquots were made so this coral will have Chl, Prot, TAC, cell density.    
- 1327: gained mass in AFDW protocol (not possible); not enough homogenate to do again.  
- 1488: homogenate tube broke (froze and cracked) and we lost half of tissue homogenate. There was not enough left to do AFDW. There will be data for Chl, Prot, TAC, cell density.

Need to re-do #2863. "Host" written down twice in notebook.

**AFDW sample size differences in addition to the 3 above sample size tables.**

| Species     	| Timepoint 	| Treatment 	| Sample Size 	| Notes                                           	|
|-------------	|-----------	|-----------	|-------------	|-------------------------------------------------	|
| P. acuta    	| Day 1     	| HTAC      	| 3           	| 1488 no data; Time limited 1st day in the field 	|
| M. capitata 	| 4 week    	| ATHC      	| 5           	| 2863 will be redone                             	|
| P. acuta    	| 8 week    	| HTAC      	| 5           	| 1327 no data                                    	|
| P. acuta    	| 8 week    	| HTHC      	| 5           	| 1174 no data                                    	|

### Chlorophyll concentration

![chl](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Chlorophyll.png?raw=true)

| Species  	| Timepoint 	| Treatment 	| Sample Size 	| Notes                                       	|
|----------	|-----------	|-----------	|-------------	|---------------------------------------------	|
| P. acuta 	| 8 week    	| HTAC      	| 5           	| 2156 missing data	|

P-2156 missing data. Label rubbed off, will need to re-do.  
Decide about 1488 sample.  

This dataset has been QC'd by Emma; final figure as of 20210118.

### Soluble Protein

![prot](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Host-Soluble-Protein.png?raw=true)

Week 4 P. acuta ATHC big drop, but samples are not in outlier tests. This still looks weird to me?

No additional groups with a decreased sample size.  
Decide about 1488 sample.      
This dataset has been QC'd by Emma; final figure as of 20210118.

### Total Antioxidant Capacity

![TAC](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/Total-Antioxidant-Capacity.png?raw=true)

No additional groups with a decreased sample size.  
Decide about 1488 sample.

Week 4 ATHC P acuta points look wonky like the protein values. Day 2 and Week 1 have very small sample sizes likely leading to large variation above?

October 7th (3 plates) were done with an incorrectly diluted stop solution. Decide if need to be re-done.  

Still having an issue with the standard curve variation. According to the [assay kit protocol](https://www.cellbiolabs.com/sites/default/files/STA-360-total-antioxidant-capacity-assay-kit.pdf), 1 mM concentration should be around 1.35 Abs490. The closest st. curve was from Nov 10th plate. All points have been projected onto the st. curve from Nov 10th and TAC calculated from this st. curve. We still need to address this.

TAC calculation done by:  

1. UAE * 2189. Convert to CRE  
2. CRE.umol.L (from Step 1) * (homog vol/1000)  
3. CRE.umol.mgprot = CRE.umol.L (from Step 2) / (prot ug / 1000)

```
TAC <- left_join(TAC.values, metadata) %>%
   mutate(cre.umol.L = uae.mM * 2189) %>%   # Convert to CRE (see product manual) per unit sample volume
   mutate(cre.umol = cre.umol.L * (vol_mL / 1000),  # Convert to CRE per coral by multiplying by homog. vol.
          cre.umol.mgprot = cre.umol / (prot_ug / 1000))  # Convert to CRE per mg protein by dividing by total protein
```

This is still producing very small values (0.04840641-1.60261156 CRE umol.mg prot). The above calculation is what was recommended by Hollie and Ross. So we have to divide by liters to get uM.mg prot to compare to the literature?


## Molecular Response Variables

Sample sizes: n=3 per treatment per timepoint.

| Species               	| Timepoint 	| Treatment 	| Sample Size 	| Notes            	|
|-----------------------	|-----------	|-----------	|-------------	|------------------	|
| M. capitata; P. acuta 	| 0 hour    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 6 hour    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 12 hour   	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 24 hour   	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 1 week    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 2 week    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 4 week    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 6 week    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 8 week    	| All       	| 3           	|                  	|
| M. capitata; P. acuta 	| 12 week   	| All       	| 3           	|                  	|
| M. capitata           	| 16 week   	| All       	| 3           	|                  	|
| P. acuta              	| 16 week   	| All       	| 2           	| Only 2 available 	|

### ITS2 Sequencing

Notebook post coming soon. Symbiont Type Profiles made from [SymPortal](https://symportal.org/). We had 17 profiles in our samples. Group label on the x-axis refers to sample. Each bar is one coral.   

Figure 20210120

![relab](https://github.com/hputnam/Acclim_Dynamics/blob/master/Output/Final_Figures/rel.abund.its2.png?raw=true)

Edits to ITS2 script confirm with Hollie, I changed:  

```
# Line 44 originally said 3:14, but that was cutting off a profile at the beginning
Pact <- as.matrix(Pact[,2:14])

# Line 76 originally said 3:14, but that was cutting off a profile at the beginning
Mcap <- as.matrix(Mcap[,2:14])

# Line 107 was [2:12], but that was cutting off a profile at the end
Rel.data <- data/rowSums(data[,2:13])
```


PERMANOVA results:  

```
# Mcap
Call:
adonis(formula = MC.mat ~ TREATMENT * TIME.POINT, data = MC.info,      permutations = 999, method = "bray")

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

                      Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)
TREATMENT              3     0.577 0.19241 0.61362 0.01419  0.766
TIME.POINT             1     0.069 0.06911 0.22041 0.00170  0.902
TREATMENT:TIME.POINT   3     1.160 0.38661 1.23293 0.02850  0.278
Residuals            124    38.883 0.31357         0.95561       
Total                131    40.689                 1.00000

# Pact
Call:
adonis(formula = PA.mat ~ TREATMENT * TIME.POINT, data = PA.info,      permutations = 999, method = "bray")

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

                      Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)
TREATMENT              3    1.1257 0.37524 1.67411 0.04156  0.137
TIME.POINT             1    0.4131 0.41314 1.84317 0.01525  0.152
TREATMENT:TIME.POINT   3    0.6673 0.22245 0.99242 0.02464  0.432
Residuals            111   24.8802 0.22415         0.91855       
Total                118   27.0865                 1.00000  

```

### mtORF Sequencing

[Notebook post](https://emmastrand.github.io/EmmaStrand_Notebook/mtORF-Amplification-Pocillopora/) for sample processing.

*P. acuta* is all one species and data visualization to come soon.

### 16s Sequencing

This is in progress in the lab at URI. Notebook post coming soon.

### Metabolomics

Coral fragments sent to Rutgers for sample processing and updates to come soon.

### Gene Expression

RNASeq analysis done by Rutgers team. Gene counts table available.

Expression of proteins (DNMTs, TETs) that are involved in DNA methylation processes.  

![mcap-epi](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Mcap.Epi.Expression.png?raw=true)

![mcap-dnmt1](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Mcap_DNMT1_expression.png?raw=true)

![mcap-dmnt3a](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Mcap_DNMT3A_expression.png?raw=true)

![mcap-tet](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Mcap_TET1_expression.png?raw=true)

![pact-epi](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Pact.Epi.Expression.png?raw=true)

![pact-dnmt1](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Pact_DNMT1_expression.png?raw=true)

![pact-dmnt3a](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Pact_DNMT3A_expression.png?raw=true)

![pact-tet](https://github.com/hputnam/Acclim_Dynamics/blob/master/Protein-Expression/Pact_TET1_expression.png?raw=true)

Figures/code HP 20201013. X-axis is old timepoint labeling system (1 = 0 hour, 3 = 6 hour, 4 = 12 hour, 5 = 24 hour, 6 = 1 week, 7 = week, 8 = 4 week, 9 = 6 week, 10 = 8 week, 11 = 12 week, 12 = 16 week). TP2 = 1 hour melted in transit and is not usable. Timepoint #s were adjusted to exclude this timepoint thereafter. The dotted line is when recovery started - this should be slightly after TP 10, not 9.  

Erin M. is also working on this, creating her own figures - code found [here](https://github.com/erinmags/Bioinformatics-Project).  

Erin C. did something similar in her MZT project - code found [here](https://github.com/echille/Montipora_OA_Development_Timeseries/blob/master/RNAseq_Analyses/MZT_biomarker_expression.Rmd) (restricted access).

Other proteins we could look into:  
Epigenetic machinery:    
- DNA methylation: DNMT1, DNMT3a   
- DNA methylation interpretation: MBD1-4, C2H2 zinc finger, SET, RING finger  
- Active demethylation: TET1-3, AID, APOBEC, TDG  
- Histone acetylation: HAT
- Histone deacetylation: Class I (nucleus): HDAC1-3,8. Class IIa (nucleus to cytosol): HDAC4,5,7,9, Class IIb (same as IIa): HDAC6,10. Class III: SIRT1,6,7 (nuclear), SIRT3,4=mitochondiral, SIRT2,5=cytosolic  
- Histone methylation: HMTs, HDMs  
- Histone demethylation: LSD1,2  

Heat shock response:  
- HSP70, 90, 60

Antioxidant response:  
- SOD, CAT, GPX, ChlAPX  


### DNA Methylation

WGBS protocol
[Notebook post]() for sample processing.
