---
title: "16S.Rmd"
author: "Emma Strand"
date: "6/10/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Load packages. 

```{r}
library(plyr)
library(dplyr)
library(tidyr)
library(readr)
library(ggplot2)

```

# 16S Lab protocol from the putnam lab 

[Notebook post](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2021-02-01-16s-Sequencing-HoloInt.md)

# Creating metadata files 

Sample manifest creation based on Hollie's that she had made prior. This will be used in the QIIME2 pathway. 
```{r}
sample.manifest <- read.csv("16S_seq/HoloInt_sample-manifest.csv") %>% dplyr::rename("sample-id" = sample.id) %>%
  dplyr::rename(`absolute-filepath` = absolute.filepath) # renaming two columns to the verbiage QIIME2 wants 
sample.manifest$`absolute-filepath` <- substr(sample.manifest$`absolute-filepath`, 55, 86) # only keep the sequence file name
sample.manifest$path <- "/data/putnamlab/estrand/HoloInt_16s/raw-data/" # adding this verbiage to each row in a new column
sample.manifest <- sample.manifest %>% unite(`absolute-filepath`, path, `absolute-filepath`, sep = "") # merging the two columns so it is the absolute filepath 

sample.manifest %>% write_csv(file = "16S_seq/ES-run/HoloInt_sample-manifest-ES.csv") # outputing to our github repo
# use the command `scp /Users/emmastrand/MyProjects/Acclim_Dynamics/16S_seq/ES-run/HoloInt_sample-manifest-ES.csv emma_strand@bluewaves.uri.edu:/data/putnamlab/estrand/HoloInt_16s/metadata` outside of andromeda to copy this file into the server 

head(sample.manifest) # double check this is the correct format of final file 
```

# See 16S bioinformatic protocol for QIIME2 steps 

[Notebook post](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2021-06-21-16s-Analysis-Pipeline.md). 

The output will be feature tables that will be used in the following section. 

## Deciding on denoising paramters for DADA2 plug-in

See notebook post link above to the parameters chosen for this comparison.
```{r}
red_short <- read.table("16S_seq/ES-run/denoising-outputs/red-short-denoising-metadata.tsv", sep="\t", header = TRUE)
  red_short$parameter <- "red_short"
red_long <- read.table("16S_seq/ES-run/denoising-outputs/red-long-denoising-metadata.tsv", sep="\t", header = TRUE)
  red_long$parameter <- "red_long"

green_short <- read.table("16S_seq/ES-run/denoising-outputs/green-short-denoising-metadata.tsv", sep="\t", header = TRUE)
  green_short$parameter <- "green_short"
green_long <- read.table("16S_seq/ES-run/denoising-outputs/green-long-denoising-metadata.tsv", sep="\t", header = TRUE)
  green_long$parameter <- "green_long"
  
yellow_short <- read.table("16S_seq/ES-run/denoising-outputs/yellow-short-denoising-metadata.tsv", sep="\t", header = TRUE)
  yellow_short$parameter <- "yellow_short"
yellow_long <- read.table("16S_seq/ES-run/denoising-outputs/yellow-long-denoising-metadata.tsv", sep="\t", header = TRUE)
  yellow_long$parameter <- "yellow_long"

# merging all above dataframes into one large data frame with all denoising statistics
denoising.stats <- union(red_short, green_short) %>% union(green_long) %>% union(yellow_long) %>% union(yellow_short) %>%
  union(red_long)
```

Transforming dataframe to be easier to plot.
```{r}
denoising.stats.gathered <- denoising.stats[, c(1,2,3,5,6,8,10,4,7,9)] #rearranging column order for the gather command so that all three desired columns are at the end

denoise.reads <- denoising.stats.gathered %>% gather(statistic, value, 3:5)

denoising.stats.gathered <- denoising.stats.gathered %>% gather(statistic, value, 8:10)
denoising.stats.gathered$statistic <- factor(denoising.stats.gathered$statistic, levels=c("percentage.of.input.passed.filter","percentage.of.input.merged","percentage.of.input.non.chimeric"))


## average stats for yellow short 
yellow.short.avg.denoise.stats <- denoising.stats %>% subset(parameter == "yellow_short") %>%
  mutate(avg.input = mean(input),
         avg.filtered = mean(filtered),
         avg.filter.perc = mean(percentage.of.input.passed.filter),
         avg.denoised = mean(denoised),
         avg.merged = mean(merged),
         avg.merged.perc = mean(percentage.of.input.merged),
         avg.non.chimeric = mean(non.chimeric),
         avg.nc.perc = mean(percentage.of.input.non.chimeric))
```

Plotting denoising statistics.
```{r}
cols <- c("darkgreen", "lightgreen", "firebrick4", "red", "darkgoldenrod2", "gold")

DADA2 <- ggplot(data = denoising.stats.gathered, aes(x = parameter, y = value, group = parameter, color = parameter)) +
  theme_classic() + geom_boxplot() +
  facet_grid(~statistic, scales = "free") +
  theme(legend.position = "none") +
  scale_color_manual(values = cols) + 
  ylab("Percentage (%)") + 
  theme(axis.text.x = element_text(angle = 60, vjust = 1.2, hjust = 1.3)); DADA2 #Set the text angle

DADA2.reads <- ggplot(data = denoise.reads, aes(x = parameter, y = value, group = parameter, color = parameter)) +
  theme_classic() + geom_boxplot() +
  facet_grid(~statistic, scales = "free") +
  theme(legend.position = "none") +
  scale_color_manual(values = cols) + 
  ylab("# reads") + 
  theme(axis.text.x = element_text(angle = 60, vjust = 1.2, hjust = 1.3)); DADA2.reads #Set the text angle

ggsave(file="Output/denoising-statistics-DADA2.png", DADA2, width = 11, height = 6, units = c("in"))
ggsave(file="Output/denoising-statistics-DADA2-reads.png", DADA2, width = 11, height = 6, units = c("in"))
```

## Denoise stats on final QIIME2 run 

```{r}
denoise.final <- read.table("16S_seq/QIIME2_20220228/denoising-stats.tsv", sep="\t", header=TRUE) 

denoise.final.reads <- denoise.final %>% select(sample.id, input, filtered, denoised, merged, non.chimeric) %>% gather(measurement, value, 2:6)
denoise.final.reads$measurement <- factor(denoise.final.reads$measurement, levels=c("input", "filtered", "denoised", "merged", "non.chimeric"))
 
denoise.final.percent <- denoise.final %>% select(
sample.id, percentage.of.input.passed.filter, percentage.of.input.merged, percentage.of.input.non.chimeric) %>% gather(measurement, value, 2:4)
denoise.final.percent$measurement <- factor(denoise.final.percent$measurement, levels=c("percentage.of.input.passed.filter", "percentage.of.input.merged", "percentage.of.input.non.chimeric"))

reads.plot <- ggplot(denoise.final.reads, aes(x=measurement, y=value), group = type) + geom_boxplot() + ylab("# reads") + theme_bw()
percent.plot <- ggplot(denoise.final.percent, aes(x=measurement, y=value), group = type) + geom_boxplot() + ylab("% reads") + theme_bw()

ggsave(file="16S_seq/QIIME2_20220228/denoise.reads.plot.png", reads.plot, width = 11, height = 6, units = c("in"))
ggsave(file="16S_seq/QIIME2_20220228/denoise.percent.plot.png", percent.plot, width = 11, height = 6, units = c("in"))

sum(denoise.final$input)
mean(denoise.final$input)
sum(denoise.final$filtered)
mean(denoise.final$filtered)
sum(denoise.final$denoised)
mean(denoise.final$denoised)
sum(denoise.final$merged)
mean(denoise.final$merged)
sum(denoise.final$non.chimeric)
mean(denoise.final$non.chimeric)
mean(denoise.final$percentage.of.input.passed.filter)
mean(denoise.final$percentage.of.input.merged)
mean(denoise.final$percentage.of.input.non.chimeric)
```

## Sample frequency QIIME2 

```{r}
sample.freq.prefilter <- read.csv("16S_seq/QIIME2_20220228/sample-frequency-detail-prefilter.csv") 

sample.freq.prefilter %>% ggplot(aes(x=X0)) + geom_histogram(color="darkblue", fill="lightblue") + theme_bw() + xlab("read frequency per sample") +
  ylab("number of samples") + ggtitle("pre-filtering")
```




## Filtering the "yellow-short" category to just "d_Bacteria"

```{r}
dbact <- read.table("16S_seq/ES-run/yellow-short-primer/yellow-short-metadata.tsv", sep="\t", header=TRUE) %>%
  filter(Taxon == "d__Bacteria")

write.table(dbact, file='16S_seq/ES-run/yellow-short-primer/yellow-short-metadata-dBact.tsv', quote=FALSE, sep='\t', col.names = NA)
```

## Checking number of bacterial sequences for a specific sample

```{r}
unfiltered.counts <- read.csv(file = "16S_seq/ES-run/yellow-short-primer/yellow-short-unfiltered-taxa-bar-plots-level1.csv", sep = ",", header = TRUE)
sample.ids <- read.csv("16S_seq/16s-sample-ids.csv", sep = ",", header = TRUE) %>% rename(index = 'X16s_ID')

unfiltered.counts <- full_join(unfiltered.counts, sample.ids, by = "index") #%>% filter(Plug_ID == "2410")

unfiltered.counts <- unfiltered.counts %>% 
  mutate(total = d__Eukaryota+d__Bacteria+Unassigned+d__Archaea) %>%
  mutate(bact_ratio = d__Bacteria/total) %>%
  mutate(avg_bact_ratio = mean(bact_ratio))

levelfour.uf.counts <- read.csv(file = "16S_seq/ES-run/yellow-short-primer/yellow-short-unfiltered-taxa-bar-plots.csv", sep = ",", header = TRUE)
levelfour.uf.counts <- full_join(levelfour.uf.counts, sample.ids, by = "index") 

levelfour.uf.counts <- levelfour.uf.counts %>% select(index:d__Bacteria.p__Firmicutes.c__Bacilli.o__Erysipelotrichales)
levelfour.uf.counts$total <- unfiltered.counts$total

unfiltered.counts$chloroplast <- levelfour.uf.counts$'d__Bacteria.p__Cyanobacteria.c__Cyanobacteriia.o__Chloroplast'

unfiltered.counts <- unfiltered.counts %>%
  mutate(chl_ratio = chloroplast/total) %>%
  mutate(avg_chl_ratio = mean(chl_ratio))

```






# Phyloseq 

R package documentation [here](https://joey711.github.io/phyloseq/). 

### Load packages 

```{r}
#BiocManager::install("phyloseq")
library('phyloseq')
library('ggplot2')
library('Rmisc')
library('cowplot')
library('ggh4x')
library('ggtext')
```

### Import dataframe with sample information and qiime2 abundance file output. 

See notes in the 16S bioinformatic pipeline for how to download this file from qiime2 view. 

```{r}
#import dataframe holding sample information
meta <- read.csv("Environmental_data/Master_Fragment.csv", header=T, sep=",", na.string="NA") %>% 
  filter(ANALYSIS == "Physiology/Molecular" | ANALYSIS == "Molecular") %>%
  select(Plug_ID, Species, Timepoint, Temperature, Treatment, CO2) 
sample.ids <- read.csv("16S_seq/16s-sample-ids.csv") %>% rename(seq.ID = X16s_ID) 

meta <- full_join(meta, sample.ids, by = "Plug_ID") %>% na.omit()

# QIIME2 output file 
abundance <- read.csv("16S_seq/abundance-level-2.csv") %>% rename(seq.ID = index) # loading in dataframe and renaming the index column to seq.ID
rownames(abundance) <- abundance$seq.ID # changing the rownames to the seq.id of that column; needed for as.matrix command
abundance.matrix <- as.matrix(abundance[,c(2:51)]) # creates a matrix out of the dataframe above 
### level 2 columns = 2:51
### level 3 columns = 2:105

head(sample.ids) # view first five lines of the dataframe 
head(abundance) # view first five lines of the dataframe 
```

### Calculating relative abundance

```{r}
Relative.abundance <- abundance.matrix/rowSums(abundance.matrix) # calculate the Relative Abundance
N <- rowSums(Relative.abundance) # calculate row sums of the relative abundnaces, should sum to 1
N #Display Row Sums, should all be 1

# calculate the Relative Abundance
Relative.abundance <- abundance[,2:51]/rowSums(abundance[,2:51]) ## exclude the first column because it is not an integer 
Relative.abundance$seq.ID <- rownames(Relative.abundance)

# merging with metadata
Relative.abundance.meta <- merge(Relative.abundance, meta, by="seq.ID")
```

### Transforming dataframe 

The goal is to have one column with relative abundance value and one column with the OTU name.

```{r}
# to get the first and last column names do colnames(abundance) in the console and copy paste column 2 and the last d__bacteria column
df <- Relative.abundance.meta %>% dplyr::group_by(Timepoint, Species, Treatment) %>% mutate(seq.ID = row_number()) %>%
  pivot_longer(cols = d__Bacteria.p__Proteobacteria:d__Archaea.p__Euryarchaeota,
               names_to = c("OTU"),
               values_to="rel.abun")

df <- df %>% filter(!Plug_ID == "2153") # removing 2153 out of the analysis because it is not the right timepoint 

df$Timepoint <- factor(df$Timepoint, levels=c("0 hour","6 hour","12 hour","30 hour","1 week","2 week","4 week", "6 week", "8 week", "12 week", "16 week"))

df$Group <- paste(df$Timepoint, df$seq.ID, sep="_")
df$Group <- as.factor(df$Group)

df$Group <- factor(df$Group, levels=c("0 hour_1", "0 hour_2", "0 hour_3",
                                      "6 hour_1", "6 hour_2", "6 hour_3",
                                      "12 hour_1", "12 hour_2", "12 hour_3",
                                      "30 hour_1", "30 hour_2", "30 hour_3",
                                      "1 week_1", "1 week_2", "1 week_3",
                                      "2 week_1", "2 week_2", "2 week_3",
                                      "4 week_1", "4 week_2", "4 week_3", 
                                      "6 week_1", "6 week_2", "6 week_3", 
                                      "8 week_1", "8 week_2", "8 week_3", 
                                      "12 week_1", "12 week_2", "12 week_3", 
                                      "16 week_1", "16 week_2", "16 week_3"))

df <- df %>% mutate(Species = case_when(
    Species == "Mcapitata" ~ "Montipora capitata",
    Species == "Pacuta" ~ "Pocillopora acuta",
  ))
```

### Plotting

```{r}
ggplot(data = df, mapping = aes(x = Group, y = rel.abun, fill = OTU)) +
    geom_bar(position="fill", stat="identity") +
    facet_wrap(facets =  vars(Species, Treatment), ncol = 4) +
    ylab("Relative Abundance") +
    theme_classic() +
    theme(axis.text.x = element_text(size=rel(0.3),angle = 90)) +
    guides(shape = guide_legend(override.aes = list(size = 0.2))) +
    guides(color = guide_legend(override.aes = list(size = 0.2))) + 
    theme(legend.position = "none")
    #theme(legend.title = element_text(size = 6), 
               #legend.text = element_text(size = 4))

#### come back to this; it wants all colors defined but there are way too many levels to do that 
bubble.plot.16s = 
  ggplot(df, aes(x = Timepoint, y = OTU)) + 
  geom_point(aes(size = rel.abun, fill = Profile), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,17), breaks = c(1,25,50,100)) + 
  facet_nested_wrap(facets =  vars(Species, Treatment), ncol = 4, nest_line = TRUE) + 
  labs(x = "", y = "", size = "Relative Abundance (%)", fill = "")  + 
  # theme(legend.key=element_blank(), 
  #       #strip.background = element_rect(color="black", fill="white", size=0.5, linetype="solid"),
  #       strip.background = element_blank(),
  #       ggh4x.facet.nestline = element_line(colour = "black"),
  #       strip.text = element_markdown(size = 12, lineheight = 1.1),
  #       axis.text.x = element_text(colour = "black", size = 12, angle = 90, vjust = 0.3, hjust = 1), 
  #       axis.text.y = element_text(colour = "black", size = 11), 
  #       legend.text = element_text(size = 10, face ="bold", colour ="black"), 
  #       legend.title = element_text(size = 12, face = "bold"), 
  #       panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 0.5), 
  #       legend.position = "bottom") +  
  #scale_fill_manual(values = colours, guide = FALSE) + 
  ggtitle("All individuals projected n=3") + 
  scale_y_discrete(limits = rev(levels(df$Profile))); bubble.plot.16s 
```



Filter out sample #2153
Timepoint values changed for those phys/molec



