---
title: "Introduction to R for microbial ecologists"
author: "Daniel Vaulot"
date: '`r format(Sys.time(), "%d %m %Y")`'
output:
  pdf_document:
    toc: yes
    number_sections : yes
    toc_depth: 2
  rmdformats::readthedown:
    highlight: kate
    number_sections : yes   
---
```{r knitr_init, echo=FALSE, cache=FALSE, warning=FALSE}
library(knitr)
library(rmdformats)
library("kableExtra")

## Global options
# The following line is necessary to make sure that
# everything is not reevaluated each time the file is knit

options(max.print="75")
  knitr::opts_chunk$set(fig.width=8,
                        fig.height=6,
                        eval=TRUE,
                        cache=TRUE,
                        echo=TRUE,
                        prompt=FALSE,
                        tidy=TRUE,
                        comment=NA,
                        message=FALSE,
                        warning=FALSE)
opts_knit$set(width=75)
```

# Aim

This document introduces basic R functions that can be used by microbial ecologists.

# Prerequisites

* Download from [GitHub](https://github.com/vaulot/R_tutorials/archive/master.zip) the whole set of tutorial

* Unzip the files to a folder on your computer

* Install [R](https://pbil.univ-lyon1.fr/CRAN/)

* Install [R studio](https://www.rstudio.com/products/rstudio/download/#download)

* Once R and R installed start R Studio and download and install the following libraries by running under R studio the following lines

```R
install.packages("dplyr")     # To manipulate dataframes
install.packages("tidyr")     # To manipulate dataframes
install.packages("readxl")    # To read Excel files into R
install.packages("ggplot2")   # for high quality graphics
install.packages("maps")      # to make maps
install.packages("treemap")   # for treemaps
install.packages("FactoMineR") # multivariate analysis

source("https://bioconductor.org/biocLite.R")
biocLite("Biostrings")        # manipulate sequences
```

# Ressources  

## Books  
\begin{center}
\includegraphics[height=200px]{../img/R_nutshell}
\includegraphics[height=200px]{../img/R_graphics_cookbook}
\end{center}

* [R-intro.pdf](https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf) : Very good introduction to R, short and clear
* [R_in_a_nutshell.pdf](http://detritus.fundacioace.com/pub/books/r_in_a_nutshell.pdf) : Many many receipes to solve all your questions
* R graphics cook book : very good for ggplot2

## Web  

### Introductory
* [Quick-R, very simple](http://www.statmethods.net/)  
* [Maps](http://www.molecularecologist.com/2012/09/making-maps-with-r/)
* [Minimal R](http://www.riffomonas.org/minimalR/)

\includegraphics[height=200px]{../img/web_quickR}

### To go one step beyond
* [Modern Dive : An Introduction to Statistical and Data Sciences via R](http://moderndive.com/index.html)
* [R for Data Science](http://r4ds.had.co.nz/)
* [Fundamentals of data visualization](http://serialmentor.com/dataviz/histograms-density-plots.html)
* [From data to visualization](https://www.data-to-viz.com/)

## Cheat sheets  
* [R basics](http://github.com/rstudio/cheatsheets/raw/master/base-r.pdf)
* [ggplot2](https://github.com/rstudio/cheatsheets/raw/master/data-visualization-2.1.pdf)
* [dplyr](https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf)

## On line course  

* [Coursera](https://www.coursera.org/)

\includegraphics[height=200px]{../img/coursera}

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
\pagebreak  

# Step by step tutorial  

## Some important points before starting

* R is an interpreted language
* R is **case sensitive**
* R works with vectors
* Types of variables : character, real, logical, factor
* Special values : TRUE, FALSE, NA
* Types of structures : vector, matrix, list, data frame
* Directory names use the linux convention: use **/** and not \


## Start R Studio  

* Go to the tutorial folder
* Switch to the subdirectory `\introduction`
* Launch R Studio
* Four windows   
    * top-left : script files / data tables
    * bottom -left: code
    * top - left : objects
    * bottom - right : help / libraries / files  / graphics

\includegraphics{../img/R_Studio_interface}

## Load necessary libraries  
```{r libraries, results='hide', message=FALSE}
library("dplyr")          # Needed to filter tables
library("tidyr")          # Needed to reshape tables from wide to long format
library("readxl")         # To read data easily
```

## Create simple vectors and data frame

### Enter the data
Our aim here to create a small table and then to compute some simple statistics

| station | fraction | diatoms | bolidos |
| --- | --- | --- | --- | --- |
| 4 | Nano | 17165 | 2 |
| 4 | Pico | 59786 | 1293 |
| 72 | Nano | 6951 | 3 |
| 72 | Pico | 3815 | 494 |

```{r}
# We enter each column as a vector
station <- c("4","4","72","72")
fraction<-c("Nano","Pico","Nano","Pico")
diatoms<-c(17165,59786,6951,3815)
bolidos<-c(2,1293,3,494)
```

### Compute new quantities
```{r}
# Add 2 columns
strameno<-diatoms+bolidos
strameno

# Divide one column by the other
diatoms_pct<-diatoms/strameno  
diatoms_pct
```

### Compute statistics
```{r}
# mean
mean(diatoms_pct)

# standard deviation
sd(diatoms_pct)

# number of observations
length(diatoms_pct)

# quick summary
summary(diatoms_pct)
```

### Accessing subsets
```{r}
diatoms_pct[1]
diatoms_pct[1:2]
```

### Data frames
```{r}
tara<-data.frame(station, fraction, diatoms, bolidos, diatoms_pct)
tara
```

### Access individual columns
```{r}
tara$diatoms
```

### Access specific lines
```{r}
tara$diatoms[tara$station==4]
```

### Compute statistics of a specific group
```{r}
mean(tara$diatoms[tara$station==4])
```

### Computing statistics according to a factor  

This can be done at least two different ways, but you will see later that it is much easier to do with the dplyr package
```{r}
# Using the tapply function
tapply(tara$diatoms, tara$station, mean)

# Using the aggregate functions
aggregate(data=tara, diatoms~station, FUN="mean")
```

\pagebreak

## Importing data

\includegraphics{../img/Tara_Excel}

A few important points :  

* Your data must be formatted in a clean table form  
     + No blank line
     + Each column must contain data of the same type (e.g. dates)
     + Missing data can be represented by empty cells
     + Each line must contain data in ALL columns
* Column titles (the first line)
     + No space (use _)
     + Always begin by letter (not a number)
* Only import primary data, all derived data can (and must) be computed with R which makes data changes much more easy  

### The hard way - exporting from Excel to a tab-delimited file

* Open Excel file  in /data directory : R_Tara.xlsx
* Copy and Paste into text file using Notepad++
* Save as R_Tara.txt

Note : you can also export from Excel but then it must be TAB-delimited (tsv file)

```{r}
tara <- read.delim("../data/R_Tara.txt")
```

Get the name and type of all the columns - Note that strings are of type "factor"
Note that empty cells are labelled as **NA** (not available) which is a R constant
```{r}
str(tara)
```
### The easy way - Read directly Excel (readxl library)

```{r}
tara <- read_excel("../data/R_Tara.xlsx", sheet = "R Tara")
```

Get the name and type of all the columns - Note that strings are now of type "char", which is better
```{r}
str(tara)
```
\pagebreak

## Compute derived quantities and Statistics  (using dplyr library)

Compute % of Bacilliarophyta and Pelagophyceae vs Total photosynthetic
```{r}
tara <- tara %>% mutate(Baci_pct = Bacillariophyta/Photo_all*100,
                        Pela_pct = Pelagophyceae/Photo_all*100)
```

Mean and SD as a function of size fraction and depth_level
```{r}
tara_stat <- tara %>% group_by(fraction, depth_level) %>%
                      summarise(Baci_pct_mean = mean(Baci_pct),
                                Baci_pct_SD = sd(Baci_pct),
                                n=n())
tara_stat

```
\pagebreak

## Do simple X-Y plots  (using ggplot2 library)

Load the ggplot2 library
```{r , results='hide', message=FALSE}
library("ggplot2")        # To do graphics
```

X vs Y

```{r, fig.height=3, fig.width=4 }
qplot(tara_salinity,tara_temp, data=tara)
```

X vs Y with variation in color of points with size fraction

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct,Pela_pct, data=tara,color=fraction)

```
\pagebreak

X vs Y with variation in color of points with size fraction and shape with depth level

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct,Pela_pct, data=tara, color=fraction, shape=depth_level)

```

X vs Y with variation in color of points with size fraction and shape with depth level

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct,Pela_pct, data=tara,color=fraction, shape=depth_level)

```
\pagebreak

X vs Y with variation sampling_depth for color of points and shape with  with size fraction.  

Note that sampling_depth is a **continuous variable**

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct,Pela_pct, data=tara, color = sampling_depth, shape = fraction)

```

Categorical data vs y with variation in color of points with depth level

```{r, fig.height=3, fig.width=6 }
qplot(fraction,Baci_pct, data=tara, color=depth_level)
```
\pagebreak

## Other types of plots

Boxplot for the same data

```{r, fig.height=3, fig.width=6 }
qplot(fraction,Baci_pct, data=tara, color=depth_level, geom="boxplot")
```

Histogram for all the data

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct, data=tara,  geom="histogram")
```
\pagebreak

Histogram with different color for each size fraction

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct, data=tara,  fill=fraction, geom="histogram")
```

Histogram with different graphs (facets) for each size fraction and depth and change bin width

```{r, fig.height=3, fig.width=6 }
qplot(Baci_pct, data=tara,  facets=fraction~depth_level, geom="histogram",binwidth=10)
```
\pagebreak

## Tree maps (much better than Pie charts...)


```{r , results='hide', message=FALSE}
library("treemap")        # To do treemaps
```

Reshape the data in order to go from the wide format to go too the long format

```{r}
tara_tree <- tara %>% select(Sample, depth_level:fraction, Strameno_all,
                             Bacillariophyta:Raphidophyceae)  %>%
                      gather(key = Class, value = n_seq,
                             Bacillariophyta:Raphidophyceae)
```

Do a global tree map

```{r,  fig.height=5, fig.width=5}
treemap(tara_tree, index = "Class", vSize= "n_seq", title = "Read numbers")
```
\pagebreak

Split the tree map according to size fraction

```{r,  fig.height=5, fig.width=5}
treemap(tara_tree, index = c("fraction", "Class"), vSize= "n_seq", title = "Read numbers")
```
\pagebreak

## Bar graphs

### Absolute abundance

Only keep surface samples
```{r }
tara_bar <- tara_tree %>% filter((depth_level=="SUR")&(fraction=="0.8-5"))
```

Do the bar plot for absolute read numbers  
* Note : rotation of labels : `theme(axis.text.x = element_text(angle = 90, hjust = 1))`

```{r,  fig.height=5, fig.width=6}
ggplot(tara_bar, aes(x = Sample , y = n_seq, fill=Class) ) +
        geom_bar(stat = "identity") +
        theme_bw()  + ggtitle("Tara - Surface - Fraction 0.8-5 µm") +
        xlab("Samples")+ylab("Number of metabarcodes") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1))
```        
\pagebreak

### Relative abundance

Compute the relative abundance of each sequence by dividing by the total number of barcodes
```{r}
tara_bar <- tara_bar %>% mutate(n_seq_rel = n_seq / Strameno_all)
```

Do the bar plot for relative read numbers
```{r,  fig.height=5, fig.width=6}
ggplot(tara_bar, aes(x = Sample , y = n_seq_rel, fill=Class) ) +
        geom_bar(stat = "identity") +
        theme_bw() + ggtitle("Tara - Surface - Fraction 0.8-5 µm") +
        xlab("Samples")+ ylab("Fraction of metabarcodes") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

\pagebreak

## Heat maps

Note : for metabarcoding data use phyloseq package.

Select the fraction and columns (from Bacillariophyta to Raphidophyceae) to be plotted and transform to a matrix

```{r}
tara_heat <- tara %>% filter(fraction=="0.8-5") %>%
                      select(Bacillariophyta:Raphidophyceae)
tara_heat.matrix <-data.matrix(tara_heat)

# It is necessary to give names to the row for heatmap labels
row.names(tara_heat.matrix) <- tara$station[fraction=="0.8-5"]
```

Draw heatmap

```{r,  fig.height=10, fig.width=10}
heatmap(tara_heat.matrix, margins = c(20,6) )
```

\pagebreak

## Multivariate analysis (FactoMiner package)

```{r , results='hide', message=FALSE}
library("FactoMineR")     # For PCA
```

Principal component analysis (PCA)

```{r PCA , fig.height=7, fig.width=7 }

# Select only the 0.8-5 µm fraction and only the colums with phytplankon data and metadata
tara_multi<- tara %>% filter(fraction=="0.8-5")

# Define row names as "Station_Depth level" (points with be labelled by row names)
row.names(tara_multi)<-paste(tara_multi$station,tara_multi$depth_level,sep="_")

# Select only with phytoplankon data and metadata
tara_multi<- tara_multi %>% select(Bacillariophyta:Raphidophyceae, chloro_hplc:tara_salinity)

# Scale the matrix
tara_multi<- scale(tara_multi)

# Do the PCA
tara_pca<-PCA(tara_multi)
```

\pagebreak

## Maps

```{r , results='hide', message=FALSE}
library("maps")           # Maps
```

Select only surface and small fraction
```{r}
tara_map <-tara %>% filter((fraction=="0.8-5")&(depth_level=="SUR"))
```

Draw the world map and add the stations

```{r, fig.height=8, fig.width=10}
# Draw the world map
map(database = "world",fill=TRUE)

# Add stations
points(tara_map$Longitude, tara_map$Latitude, pch=3, col="red", cex=1)

# Add data - circle size is proprotional to proportion of
points(tara_map$Longitude, tara_map$Latitude,
       pch=19, col="blue", cex= tara_map$Baci_pct *3/100)

# Add title
title("Bacilliorophyta as % of Photosynthetic - 0.8-5 µm - surface",cex.main=1.0)
```

\pagebreak

## Manipulate sequences

In  BioConductor there are many packages that can process sequences either GenBank or short reads
```{r , results='hide', message=FALSE}
library("Biostrings")     # To manipulate sequences
```

Read sequences from metagenome (454)
```{r}
seq<-readDNAStringSet("../data/BIOSOPE_T142_reads_random.fasta", format="fasta")
```

Compute length of sequence (discard N), compute statistics and plot histogram
```{r, fig.height=3, fig.width=6 }
Length_seq<-letterFrequency(seq, letters="ATCG")

range(Length_seq)

mean(Length_seq)

qplot(Length_seq, geom="histogram", xlab="Sequence length", ylab="Frequency")
```

Compute GC% and do simple plots
```{r, , fig.height=3, fig.width=6}
# Compute number of "GC"
GC_seq <- letterFrequency(seq, letters="CG")

# Compute GC % in sequence
GC_percent <- GC_seq/Length_seq

# Do histogram
qplot(GC_percent, geom="histogram", xlab="GC %", ylab="Frequency")

# Plot GC % vs Length of sequence
qplot(GC_percent,Length_seq)
```


**Exercice** : Load sequence from Bathycoccus and compare GC% to that of the whole metagenome
```{r}
seq <- readDNAStringSet("../data/BIOSOPE_T142_reads_Bathy.fasta", format="fasta")
```

\pagebreak

\begin{center}
\includegraphics{../img/achille_talon}

\textbf{Your turn now. These are just a few of the things you can do, possibilities are endless...}
\end{center}
